### PR TITLE
[CIT-374] c8y mapper threshold packetsize

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

This PR captures the code changes for validating the packet size of c8y mapper.
Since the c8y cloud does not accept packets that are bigger than the 16KB, the packets that are above this size will be dropped.

This PR also adds a pysys system test as well as a rust integration test.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    let (mut router, _console, servers, _builder) = async_locallink::construct_broker(config);
+    let router = tokio::task::spawn_blocking(move || -> anyhow::Result<()> { Ok(router.start()?) });
+    servers.await;
+    let _ = router.await;
+    Ok(())
+}
+
+async fn publish_big_message_wait_for_error() -> Result<(), anyhow::Error> {
+    // create a 20KB message
+    let buffer = create_packet(1024 * 20);
```
Comment (makr11st) [inline mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs:71]: I believe I already saw this functions somewhere...
Maybe you could move it to `tedge_utils` crate?

File: mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        .args(&["tedge", "config", "unset", "mqtt.port"])
+        .output()
+        .expect("failed to execute process");
+
+    // stop tedge mapper
+    let _sysctl_output = Command::new("sudo")
+        .args(&["systemctl", "stop", "tedge-mapper-c8y.service"])
+        .output()
+        .expect("failed to execute process");
+}
```
Comment (makr11st) [inline mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs:150]: I understand the need for the setup function, but I'm struggling here with the need to call `tedge cli` with sudo? Is there a problem to create a configuration structure by hand in this function?
`TedgeConfig` is what you are looking for.

File: tests/PySys/tedge_mapper_c8y/tedge_mapper_c8y_packet_threshold_size/run.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Given a configured system
+When we start the tedge-mapper-c8y systemctl service
+When we start tedge sub with sudo in the background for error messages
+When we publish a bigger message
+Then we kill tedge sub with sudo as it is running with a different user account
+Then we validate the error message in the output of tedge sub
+Then we stop the tedge-mapper-c8y systemctl service
+
+"""
```
Comment (makr11st) [inline tests/PySys/tedge_mapper_c8y/tedge_mapper_c8y_packet_threshold_size/run.py:17]: I believe for this story this test is redundant unless we want to test end to end behaviour as well.

**Review by makr11st — CHANGES_REQUESTED** [review]
There is no need to change your system configuration files to create a mapper with specific test config. 
Eg, you can use TedgeConfig struct to do so without reading it from file.

File: mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs (hunk 1 above)
Comment (PradeepKiruvale) [reply mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs]: Sure, this is already under another refactoring PR. I will do it as part of that.

File: tests/PySys/tedge_mapper_c8y/tedge_mapper_c8y_packet_threshold_size/run.py (hunk 3 above)
Comment (PradeepKiruvale) [reply tests/PySys/tedge_mapper_c8y/tedge_mapper_c8y_packet_threshold_size/run.py:17]: This is kind of a black box test that tests the whole c8y_mapper. So, I feel it's better to keep it.

**Review by makr11st — APPROVED** [review]

**Pull request merged**